crates/ruff_linter/src/rules/flake8_return/rules/function.rs

```diff
@@ -391,6 +390,25 @@ fn unnecessary_return_none(checker: &mut Checker, decorator_list: &[Decorator],
     }
 }
 
+/// Determine whether `decorator` is `@property`,
+/// or another stdlib decorator similar to `@property.
+///
+/// TODO: be more principled here once we have type inference;
+/// hardcoding these is a little hacky.
+fn is_property_like_decorator(decorator: &Decorator, semantic: &SemanticModel) -> bool {
+    semantic
+        .resolve_qualified_name(&decorator.expression)
```

Comment:
@epenet -- you had `semantic.resolve_qualified_name(map_callable(&decorator.expression))` here. I think using `map_callable` would be incorrect in this instance. `map_callable()` is a convenience function for when we don't care whether e.g. a function is decorated with `@functools.lru_cache` or `@functools.lru_cache()` (whether the decorator is a call-expression or not is irrelevant). But for all of these `property`-like decorators, you can only use them as `@property`; it's invalid to use them as `@property()`. So for these decorators, we just want to call `resolve_qualified_name` directly on the expression, instead of passing it through `map_callable` first.

Reply:
I copied it from elsewhere in the code base.
I wonder if this applies there as well...
And would it make sense to move this function to a shared location?

Reply:
> I copied it from elsewhere in the code base.
> I wonder if this applies there as well...

It's hard to say in general. For a lot of decorators (like `@lru_cache`), you really don't care about whether it's been decorated with `@lru_cache` or `@lru_cache()`. So while there may be other places in the codebase where it's being used incorrectly, I'm pretty confident they're not _all_ incorrect ;-)

> And would it make sense to move this function to a shared location?

Possibly...

Reply:
Good that you asked. It seems we already have a general-purpose function for this:

https://github.com/astral-sh/ruff/blob/aaa56eb0bd7f2740c83587d03198bc12c8e69648/crates/ruff_python_semantic/src/analyze/visibility.rs#L63-L83

Reply:
For reference, I wonder if it might make sense to look at `rules/pydoclint/rules/check_docstring.rs` also:
https://github.com/astral-sh/ruff/blob/459c85ba273ae347402b1759cacd1e7740cef121/crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs#L442-L461

Reply:
nice, thanks